### ISSUE TITLE
SplitButton: aria-expanded is not updated when clicking outside

### ISSUE THREAD

**Issue by tri-harmoney** [primefaces/primeng#15710]
### Describe the bug

The `aria-expanded` attribute is not updated correctly if the menu is closed by clicking outside.

If the split menubutton is clicked the menu opens and the `aria-expanded` is set to `true`. If the user now clicks outside or presses esc button, the menu closes but the `aria-expanded` is not set to `false` but stays `true`.

![image](https://github.com/primefaces/primeng/assets/112377329/c91ea252-49bf-485c-8404-4c6c0114e446)

### Environment

macOS 14.5 (23F79)

### Angular version

17.3.7

### PrimeNG version

17.16.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.13.0

### Steps to reproduce the behavior

1. Click the menu open through the menubutton
2. Close the menu by pressing ESC button or clicking outside the element
3. Check `aria-expanded`

### Expected behavior

`aria-expanded` should be set to false.
